### ISSUE TITLE
Specify behavior of async resource detection

### ISSUE THREAD

**Issue by dyladan** [open-telemetry/opentelemetry-specification#952]
**What are you trying to achieve?**

tl;dr: what happens if spans are created while resources are still being detected?

In some environments such as GCP, some resource attributes are detected by calling a remote endpoint. In JS at least, it is impossible to make blocking IO calls. This means the user app either has to wait for the resource detection to complete before starting the SDK, or start the SDK and potentially start creating spans before the resource detection is complete. I understand in other languages like Java it is possible to block, but you can't block all threads and some spans might be created in another thread while the resources are still being detected.

I asked this question in the maintainers gitter channel and according to @jkwatson the current behavior in Java is to drop these spans. This may be an issue in latency sensitive contexts like lambda, where the app quickly boots and creates some spans. In many cases, those spans might be the only spans. According to @tsloughter, the erlang currently does not allow for remote/async/delayed resources.

I don't know if this directly affects the trace GA, but I feel this is an important thing to get nailed down.

**Comment by dyladan** [comment]
From spec meeting:

Since resources are not API surface and are SDK only, this is not a GA blocker. Because this is a common scenario that everyone will hit eventually, it is important that we specify something eventually. The current SDK does not do resource auto-detection, and depends on "extension packages," which allows us to offload these concerns from the SDK for the moment.

**Comment by svrnm** [comment]
The label removal we applied during triage stopped the `stale` bot from closing this issue, so we close it manually, if this is incorrect please re-open or raise a new issue, cc @mx-psi @jpkrohling
